app/services/katello/pulp3/repository.rb

```diff
@@ -268,6 +268,20 @@ def self.orphan_repository_versions(smart_proxy)
         version_hrefs - ::Katello::Repository.where(version_href: version_hrefs).pluck(:version_href)
       end
 
+      def self.orphan_repository_versions_for_mirror(smart_proxy)
+        current_pulp_repositories = ::Katello::Pulp3::Repository.list(smart_proxy, {})
+        repo_hrefs = current_pulp_repositories.collect { |repo| repo[:_href] }.uniq
+
+        version_hrefs = repo_hrefs.collect do |href|
+          repo_versions = ::Katello::Pulp3::Repository.versions_list(
+            smart_proxy, href, ordering: :_created)
+          repo_versions.pop
```

Comment:
instead of just removing the last item, it might be better to compare it to the 'latest_version' href on the repository object returned by pulp

Reply:
Good point. Is there ever a chance the latest_version wouldn't actually be the "latest" version on a mirror? Would it matter if it wasn't?

Reply:
i don't think it would, it should be updated on every sync, or anytime a new version is created.